### PR TITLE
Updated to Gradle 7 (version 7.4.2)

### LINKED ISSUE #662: Update to Gradle 7
**Is your feature request related to a problem? Please describe.**
The OpenSearch Plugins are still built using Gradle 6.6.x whereas the Gradle release train has moved to 7.x already. The OpenSearch core is about to be switched to Gradle 7.3 (see please opensearch-project/OpenSearch#1609), it would make sense to switch all plugins to Gradle 7.3 as well.

**Describe the solution you'd like**
 Update Gradle to 7.3

**Describe alternatives you've considered (Optional)**
N/A

**Additional context**
See please opensearch-project/OpenSearch#1246 and opensearch-project/opensearch-plugins#107

### PR DESCRIPTION
**Pull request by dlvenable**
### Description

Updated the project to Gradle 7.
 
### Issues Resolved

Resolves #662 
 
### Check List
- [x] New functionality includes testing.
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,11 +1,5 @@
-#
-# Copyright OpenSearch Contributors
-# SPDX-License-Identifier: Apache-2.0
-#
```
Comment (asifsmohammed) [inline gradle/wrapper/gradle-wrapper.properties:4]: Isn't the copyright required for .properties files?

**Review by asifsmohammed — APPROVED** [review]

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (dlvenable) [reply gradle/wrapper/gradle-wrapper.properties:4]: I don't believe we should have it here because this is a file which was generated by Gradle itself.

Comment (dlvenable) [reply gradle/wrapper/gradle-wrapper.properties:4]: Though, it does appear that the OpenSearch core project includes it.

https://github.com/opensearch-project/OpenSearch/blob/c41b41140b745fa62b38afe7ead27002d77e809f/gradle/wrapper/gradle-wrapper.properties#L1-L10

Comment (dlvenable) [reply gradle/wrapper/gradle-wrapper.properties]: @asifsmohammed , I added the header back to match what OpenSearch core does.

**Review by asifsmohammed — APPROVED** [review]

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -3,9 +3,8 @@
 # SPDX-License-Identifier: Apache-2.0
 #
 
-#Mon Aug 03 12:43:56 CDT 2020
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.6.1-all.zip
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-zipStorePath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.4.2-bin.zip
```
Comment (sbayer55) [inline gradle/wrapper/gradle-wrapper.properties:8]: Thanks for switching to `bin`

**Review by sbayer55 — APPROVED** [review]

**Pull request merged**